ouroboros-consensus/src/Ouroboros/Consensus/BlockchainTime/API.hs

```diff
@@ -41,16 +42,20 @@ data BlockchainTime m = BlockchainTime {
       -- itself was created.
       --
       -- Use sites should call 'onSlotChange' rather than 'onSlotChange_'.
-    , onSlotChange_  :: HasCallStack => (SlotNo -> m ()) -> m (m ())
+    , onSlotChange_  :: HasCallStack
+                     => ResourceRegistry m -> (SlotNo -> m ()) -> m (m ())
```

Comment:
> now we _implicitly_ require (B) to outlive (A)

That doesn't seem necessary to me. `onSlotChange` fires when the slot (i.e. `TVar`) changes -- if A outlives B, then the slot will stop changing. That seems like an expected behavior. Am I being myopic?

Reply:
Some what related... I've been anticipating that clock skew will require multiple `btime`s. I had also anticipated that that if we ever want no clock skew (e.g. after shrinking), then it'd be ideal if we could key off of a single `btime`/`TVar`. But I suppose that doesn't actually matter in `io-sim`, since artificial synchrony is essentially possible there.

Reply:
I reverted this change. I added `cloneTestBlockchainTime` to emphasize that all of the `btime`s are still perfectly synchronized for now; it just overrides the `onSlotChange_` field to use the newly given registry.